pkg/controller/combo_controller.go

```diff
@@ -153,6 +155,41 @@ func (c *combinationController) Reconcile(ctx context.Context, req ctrl.Request)
 		Message:            "evaluations successfully processed",
 	})
 
+	// If manifest application is indicated, create generated manifests on-cluster.
+	if combination.Spec.Apply{
+		for _, manifest := range generatedManifests {
+			dec := yaml.NewYAMLOrJSONDecoder(strings.NewReader(manifest), 10)
+			obj := &unstructured.Unstructured{}
+			err := dec.Decode(obj)
+			if err != nil {
+				combination.SetStatusCondition(metav1.Condition{
+					Type:               v1alpha1.TypeInvalid,
+					Status:             metav1.ConditionFalse,
+					Reason:             v1alpha1.ReasonCreationFailed,
+					LastTransitionTime: metav1.NewTime(time.Now()),
+					Message:            fmt.Sprintf("failed to decode manifest %s to create on-cluster: %s", obj.GetName(), err.Error()),
+				})
+			}
+			err = c.Create(ctx, obj)
+			if err != nil {
+				combination.SetStatusCondition(metav1.Condition{
+					Type:               v1alpha1.TypeInvalid,
+					Status:             metav1.ConditionFalse,
+					Reason:             v1alpha1.ReasonCreationFailed,
+					LastTransitionTime: metav1.NewTime(time.Now()),
+					Message:            fmt.Sprintf("failed to create manifest %s on-cluster: %s", obj.GetName(), err.Error()),
+				})
+			}
+			combination.SetStatusCondition(metav1.Condition{
+				Type:               v1alpha1.TypeFinished,
+				Status:             metav1.ConditionTrue,
+				Reason:             v1alpha1.ReasonCreationSucceeded,
+				LastTransitionTime: metav1.NewTime(time.Now()),
+				Message:            fmt.Sprintf("created manifest %s on-cluster", obj.GetName()),
+			})
+		}
+	}
```

Comment:
What are your thoughts on this? This seems to make a bit more sense to me as we'd want to update with a successful reason once all manifests have been processed (since there are few applies happening in that loop):

```suggestion
			if err != nil {
				combination.SetStatusCondition(metav1.Condition{
					Type:               v1alpha1.TypeInvalid,
					Status:             metav1.ConditionFalse,
					Reason:             v1alpha1.ReasonCreationFailed,
					LastTransitionTime: metav1.NewTime(time.Now()),
					Message:            fmt.Sprintf("failed to create manifest %s on-cluster: %s", obj.GetName(), err.Error()),
				})
			}
		}
		combination.SetStatusCondition(metav1.Condition{
				Type:               v1alpha1.TypeFinished,
				Status:             metav1.ConditionTrue,
				Reason:             v1alpha1.ReasonCreationSucceeded,
				LastTransitionTime: metav1.NewTime(time.Now()),
				Message:            fmt.Sprintf("created manifest %s on-cluster", obj.GetName()),
			})
	}
```